### DISCUSSION TITLE
Keeping connections open after cancel

### DISCUSSION THREAD

**Discussion by TravisSmithBlz** [curl/curl#11056] [Q&A] [answered]
Hello friends!

I am looking for insight into the work required to add support for keeping connections open after removing an in-flight easy-handle from a multi-handle.

For context, our application often re-prioritizes what data it needs and this results in a handful of requests being cancelled and another handful being started. Due to the closing of the connections, we suffer a significant performance loss reestablishing them.

I don't mind doing this work, but I need to determine level of effort so I can weight it against possible workarounds in our application code. 

Any help would be appreciated, thank you.

Travis

**Comment by bagder** [comment, marked as answer]
libcurl should do its best to maintain connections alive since that's a primary key to doing subsequent transfers faster. All cases of unnecessary discarding a working connection is a bug to some degree.

You do not explain the nature nor protocol of the connections you are using. I then presume that you refer to HTTP/1.

For HTTP/1 connections and transfers, it is basically impossible to correctly stop a transfer and yet maintain the connection. The protocol is not designed for it.

**Reply by TravisSmithBlz** [reply]
Thank you for your time. You are correct, I was speaking of HTTP/1.
